interface.go

```diff
@@ -0,0 +1,19 @@
+package option
+
+import (
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type genericInterface[T any] interface {
```

Comment:
It looks like `optionInterface` or `commonInterface` since it is a common interface for the package types.

Please, add a short-comment description to the type: why do we need the interface? Since it is not obvious for a code reader.